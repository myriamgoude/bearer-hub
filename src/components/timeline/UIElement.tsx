import * as React from 'react'
import GraphImg from 'graphcms-image'
import styled from '@emotion/styled'

import CodeSnippet from './CodeSnippet'

const StyledDiv = styled.div<{ rightAligned: boolean }>`
  width: 50%;
  ${props => (props.rightAligned ? 'margin-left: auto;' : '')};
`

interface IUIElementProps {
  element: {
    title: string
    codeSnippet: string
    image: {
      handle: string
      height: number
      width: number
    }
  }
  rightAligned: boolean
}

const IUIElementProps = ({ element, rightAligned }: IUIElementProps) => (
  <StyledDiv rightAligned={rightAligned}>
    <h4>{element.title}</h4>
    {element.image ? <GraphImg image={element.image} maxWidth={800} /> : null}
    <CodeSnippet>{element.codeSnippet}</CodeSnippet>
  </StyledDiv>
)

export default IUIElementProps
