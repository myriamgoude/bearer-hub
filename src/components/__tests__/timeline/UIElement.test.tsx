import React from 'react'
import * as ShallowRenderer from 'react-test-renderer/shallow'

import UIElement from '../../timeline/UIElement'

describe('UIElement', () => {
  const element = {
    title: 'Connect Hello World component',
    codeSnippet: '<hello-world-connect />',
    image: {
      handle: 'my-test-image',
      height: 50,
      width: 50
    }
  }

  it('has no auto margin when rightAligned is false', () => {
    const rightAligned = false

    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<UIElement element={element} rightAligned={rightAligned} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('has auto margin-left when rightAligned is true', () => {
    const rightAligned = false

    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<UIElement element={element} rightAligned={rightAligned} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
