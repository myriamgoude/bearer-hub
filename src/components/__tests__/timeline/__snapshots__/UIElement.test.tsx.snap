// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIElement has auto margin-left when rightAligned is true 1`] = `
<ForwardRef(render)
  rightAligned={false}
>
  <h4>
    Connect Hello World component
  </h4>
  <GraphImage
    alt=""
    backgroundColor=""
    baseURI="https://media.graphcms.com"
    blurryPlaceholder={true}
    className=""
    fadeIn={true}
    fit="crop"
    image={
      Object {
        "handle": "my-test-image",
        "height": 50,
        "width": 50,
      }
    }
    maxWidth={800}
    onLoad={null}
    outerWrapperClassName=""
    style={Object {}}
    title=""
    transforms={Array []}
    withWebp={true}
  />
  <CodeSnippet>
    &lt;hello-world-connect /&gt;
  </CodeSnippet>
</ForwardRef(render)>
`;

exports[`UIElement has no auto margin when rightAligned is false 1`] = `
<ForwardRef(render)
  rightAligned={false}
>
  <h4>
    Connect Hello World component
  </h4>
  <GraphImage
    alt=""
    backgroundColor=""
    baseURI="https://media.graphcms.com"
    blurryPlaceholder={true}
    className=""
    fadeIn={true}
    fit="crop"
    image={
      Object {
        "handle": "my-test-image",
        "height": 50,
        "width": 50,
      }
    }
    maxWidth={800}
    onLoad={null}
    outerWrapperClassName=""
    style={Object {}}
    title=""
    transforms={Array []}
    withWebp={true}
  />
  <CodeSnippet>
    &lt;hello-world-connect /&gt;
  </CodeSnippet>
</ForwardRef(render)>
`;
